lib/coupon_code.rb

```diff
@@ -36,15 +33,14 @@ def self.validate(orig, num_parts = PARTS)
   end
 
   def self.checkdigit_alg_1(orig, check)
-
-    orig.split('').each_with_index do |c, i|
+    orig.split('').each_with_index do |c, _|
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.